test/cmd.spec.js

```diff
@@ -227,6 +227,48 @@ describe('Cmds', () => {
       });
     });
 
+    describe('Cmd.setTimeout', () => {
+      it('resolves with the action created from the action creator', async () => {
+        let dispatchPromise = new Promise(resolve => {
+          dispatch = jest.fn(() => resolve());
+        });
+
+        let action = actionCreator1(123);
+        let cmd = Cmd.setTimeout(Cmd.action(action), 100, {
+          scheduledActionCreator: actionCreator2
+        });
+        let result = executeCmd(cmd, dispatch, getState);
+
+        await expect(result).resolves.toEqual([
+          actionCreator2(expect.anything())
+        ]);
+        await dispatchPromise;
```

Comment:
shouldn't we mock the timers and wait for those rather than just resolve as soon as dispatch is called? That would allow us to verify we're actually waiting for the right amount of time, and would allow us to improve the setInterval test by verifying it actually runs multiple times

Reply:
Yes, do you have a nice way to do this, or would you just overwrite the global `setInterval` and restore it in an after block.

Reply:
jest can do this https://jestjs.io/docs/en/timer-mocks

Reply:
Done, please check it again.

Reply:
can you do the same thing for setTimeout? basically the same as the setInterval test, except we verify that we don't dispatch more actions after the time elapses again

Reply:
Done.